frostsnap_coordinator/src/bitcoin/wallet.rs

```diff
@@ -487,22 +349,21 @@ impl FrostsnapWallet {
         }
 
         if let Some(value) = cs.drain_value(target, change_policy) {
-            let ((i, _change_spk), changeset) = self
-                .graph
-                .index
-                .next_unused_spk(&(key_id, AppAccountKeychain::internal()))
-                .expect("this should have been initialzed by now since we are spending from it");
-            self.db
-                .lock()
-                .unwrap()
-                .execute(|tx| {
-                    self.consume_spk_revelation_change(tx, changeset)?;
-                    Ok(())
-                })
-                .context("trying to persist change derivation increment")?;
-            self.graph
+            let mut db = self.db.lock().unwrap();
+            let (i, _change_spk) = self.tx_graph.mutate(&mut *db, |tx_graph| {
+                Ok(tx_graph
+                    .index
+                    .next_unused_spk(&(key_id, AppAccountKeychain::internal()))
+                    .expect(
+                        "this should have been initialzed by now since we are spending from it",
+                    ))
+            })?;
+
+            self.tx_graph
+                .MUTATE_NO_PERSIST()
```

Comment:
Curious why MUTATE_NO_PERSIST here?

Because we don't want to mark as used until later (tx broadcasted / chain synced)?

Reply:
good question. It demonstrates why this `MUTATE_NO_PERSIST` has to be fixed long term. The answer is that `mark_used` doesn't persist anything but how are you meant to know that.